ui/pages/confirmations/components/confirm/footer/footer.tsx

```diff
@@ -18,6 +19,129 @@ import {
 } from '../../../../../store/actions';
 import { confirmSelector } from '../../../selectors';
 import { getConfirmationSender } from '../utils';
+import useAlerts from '../../../../../hooks/useAlerts';
+import { MultipleAlertModal } from '../../../../../components/app/confirmations/alerts/multiple-alert-modal';
+import { AlertModal } from '../../../../../components/app/confirmations/alerts/alert-modal';
+
+function ConfirmAlertModal({
+  alertKey,
+  ownerId,
+  handleCloseModal,
+  hasUnconfirmedAlerts,
+  alertModalVisible,
+  frictionModalVisible,
+  handleOpenModal,
+  onCancel,
+  onSubmit,
+}: {
+  alertKey: string;
+  ownerId: string;
+  handleCloseModal: () => void;
+  hasUnconfirmedAlerts: boolean;
+  alertModalVisible: boolean;
+  frictionModalVisible: boolean;
+  handleOpenModal: () => void;
+  onCancel?: () => void;
+  onSubmit?: () => void;
+}) {
+  if (alertModalVisible) {
+    return (
+      <MultipleAlertModal
+        alertKey={alertKey}
+        ownerId={ownerId}
+        onFinalAcknowledgeClick={handleCloseModal}
+        onClose={handleCloseModal}
+      />
+    );
+  }
+
+  if (!hasUnconfirmedAlerts && frictionModalVisible) {
+    return (
+      <AlertModal
+        alertKey={alertKey}
+        ownerId={ownerId}
+        onAcknowledgeClick={handleCloseModal}
+        onClose={handleCloseModal}
+        frictionModalConfig={{
+          onAlertLinkClick: handleOpenModal,
```

Comment:
The logic to display the multiple alert modal is currently in here, but would it be simpler to encapsulate that within the new component I suggested, since it's core purpose is to let the user review the alerts with the button and do the final acknowledge.

It would still need an `onCancel` or `onSubmit` though.

Reply:
I agree, transfered all logic around displaying the `MultipleAlertModal` to the new `ConfirmAlerModal`.